### ISSUE TITLE
Checkbox group binding does not update on programmatic change after user interaction

### ISSUE THREAD

**Issue by brunnerh** [carbon-design-system/carbon-components-svelte#967]
If the bound value is changed the components do not necessarily update and the checking state becomes desynchronized (unchecking the item causes the item value to be added to the group).

Repro steps:
1. Add `Checkbox` components with `bind:group`.
2. Add `Button` that changes underlying bound array.
3. Click various checkboxes and click button to cause programmatic change.

If the checkboxes have not been interacted with yet, the programmatic change works.

[REPL](https://svelte.dev/repl/351cd005796f4e23842be1458c529b8f?version=3.44.3)

**Comment by brunnerh** [comment]
This would fix most issues, but the variables bound to `checked` could no longer be changed while using `group`: https://github.com/brunnerh/carbon-components-svelte/commit/a67a41b8832e8d923457d0346edd3ea07c1b5008

**Comment by metonym** [comment]
@brunnerh As always, I really appreciate the extensive thought you put into this.

#1042 should address the reactivity bug by only using one variable: [`checked`](https://github.com/carbon-design-system/carbon-components-svelte/pull/1042/files#diff-2ec9a77a035afc5a6489a47b3980639426bf70c891a3f9a773f69b522e741c10R63).

I've tested the following use cases:

```svelte
<script>
  import { Checkbox } from "carbon-components-svelte";

  let options = ["Apple", "Banana", "Coconut"];
  let selection = options.slice(0, 2);
  let selection2 = [];

  let checked = true;
  let checked2 = false;
  let checked3 = false;
</script>

<h2>bind:group</h2>
{#each options as option}
  <Checkbox bind:group={selection} labelText={option} value={option} />
{/each}
<div>
  Selected options:
  <strong>{selection.join(", ")}</strong>
</div>
<button on:click={() => (selection = ["Banana"])}>Set to Banana</button>

<h2>bind:group with checked</h2>
<Checkbox bind:group={selection2} labelText="Apple" value="Apple" />
<Checkbox
  bind:group={selection2}
  bind:checked={checked3}
  labelText="Banana"
  value="Banana"
/>
<Checkbox bind:group={selection2} labelText="Coconut" value="Coconut" />
<div>
  Selected options:
  <strong>{selection2.join(", ")}</strong><br />
  checked3: <strong>{checked3}</strong>
</div>
<button on:click={() => (selection2 = ["Banana"])}>Set to Banana</button>

<h2>bind:checked</h2>
<Checkbox bind:checked={checked2} />
<button on:click={() => (checked2 = !checked2)}>Set ({checked2})</button>

<h2>manual</h2>
<Checkbox {checked} on:change={(e) => (checked = e.target.checked)} />
<button on:click={() => (checked = !checked)}>Set ({checked})</button>

```

The only issue is with `bind:group` and `bind:checked`; `bind:checked` will only support one-way binding but that's an acceptable trade-off as `bind:group` is still two-way.

**Comment by brunnerh** [comment]
> The only issue is with bind:group and bind:checked; bind:checked will only support one-way binding but that's an acceptable trade-off as bind:group is still two-way.

Maybe that limitation should be documented to prevent mistakes/confusion?

**Comment by metonym** [comment]
Fixed in [v0.56.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.56.0)
